LanguageFeatures/Augmentation-libraries/augmenting_members_A03_t01.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// A non-final variable is augmented with a final variable.
```

Comment:
OK, this is one of those cases that we're discussing in https://github.com/dart-lang/language/issues/4017.

As usual, we will follow the current spec and update the test later if needed.